GenerateSacrificeSettings.sh

```diff
@@ -0,0 +1,440 @@
+# end of sound options
+
+# Multiplayer part
+echo "### multiplayer options for host ###" >> settings.txt
+
+echo "1. Host for 1v1"
+echo "2. Host for 2v2"
+echo "3. Host for 3v3"
+echo "4. Join"
+echo "5. Host FFA"
+
+hostmode=""
+while [[ ! $hostmode =~ ^[1-5]$ ]]; do
+    read -p "Enter a number between 1 and 5: " hostmode
+    if [[ ! $hostmode =~ ^[1-5]$ ]]; then
+        print_error_message
+    fi
+done
+
+## HOSTMODE
+case $hostmode in
+    # FFA
+    5)  
+		echo "1. Host for 3"
+		echo "2. Host for 4"
+		echo "3. Host for 5"
+		echo "Hosting FFA for 6 is not available due to lack of maps and maplists"
+		ffamode=""
+		while [[ ! $ffamode =~ ^[1-3]$ ]]; do
+			read -p "Enter a number between 1 and 3: " ffamode
+			if [[ ! $ffamode =~ ^[1-3]$ ]]; then
+				print_error_message
+			fi
+		done
+		ffaplayers=""
+		case $ffamode in
+			1) ffaplayers=3 ;;
+			2) ffaplayers=4 ;;
+			3) ffaplayers=5 ;;
+		esac
+
+		# Finally, echo everything to settings.txt
+		echo "--host=$ffaplayers" >> settings.txt
+		echo "--ffa                                        # host ffa (overrides team settings)" >> settings.txt
+		echo "--random-gods" >> settings.txt
+		echo "--shuffle-sides" >> settings.txt
+		echo "--stutter-on-desynch" >> settings.txt
+		echo "--no-particles" >> settings.txt
+		# if there's only 3 ffa players, offer another choice, otherwise use tg-{$ffaplayers}ffa.txt maplist
+		if [ $ffaplayers != 3 ]; then
+			echo "--map-list=maps-tg-{ffaplayers}ffa.txt" >> settings.txt
+		else
+			# another maplist choice
+			echo "1. shiny's 3ffa map list"
+			echo "2. tg's 3ffa map list"
+			echo "3. tree's 3ffa map list"
+			maplist=""
+			while [[ ! $maplist =~ ^[1-3]$ ]]; do
+				read -p "Enter a number between 1 and 3: " maplist
+				if [[ ! $maplist =~ ^[1-3]$ ]]; then
+					print_error_message
+				fi
+			done
+			case $maplist in
+				1) # shiny ffa3
+					echo "--map-list=maps-shiny-3ffa.txt" >> settings.txt
+					;;
+				2) # tg ffa3
+					echo "--map-list=maps-tg-3ffa.txt" >> settings.txt
+					;;
+				3) # tree ffa3
+					echo "--map-list=maps-tree-3ffa.txt" >> settings.txt
+					;;
+			esac
+		fi
+		;;
+    # join
+    4)
+		echo "--join" >> settings.txt
+		# call the observer() function we declared at the bottom of the file
+		observer
+		;;
+	# 3v3
+	3)
+		echo "--host=6" >> settings.txt
+		echo "--map-list=maps-tg-3v3.txt" >> settings.txt
+		echo "--3v3                                        # host 3v3 (overrides team settings)" >> settings.txt
+		echo "--random-gods" >> settings.txt
+		echo "--shuffle-sides" >> settings.txt
+		echo "--stutter-on-desynch" >> settings.txt
+		echo "--no-particles" >> settings.txt
+		;;
+	# 2v2
+	2)
+		echo "--host=4" >> settings.txt
+		echo "--2v2                                        # host 2v2 (overrides team settings)" >> settings.txt
+		echo "--random-gods" >> settings.txt
+		echo "--shuffle-sides" >> settings.txt
+		echo "--stutter-on-desynch" >> settings.txt
```

Comment:
this whole bit could be improved.

especially this part:
```
		echo "--random-gods" >> settings.txt
		echo "--shuffle-sides" >> settings.txt
		echo "--stutter-on-desynch" >> settings.txt
		echo "--no-particles" >> settings.txt
```
seems to be repeated everywhere. extract to a method, then just call that method.

Reply:
in fact, make the `--no-particles` customizable as well.

Reply:
done. resolving.